openedx_events/event_bus/avro/tests/test_avro.py

```diff
@@ -112,6 +116,16 @@ def generate_test_event_data_for_data_type(data_type):  # pragma: no cover
         datetime: datetime.now(),
         CCXLocator: CCXLocator(org='edx', course='DemoX', run='Demo_course', ccx='1'),
         UUID: uuid4(),
+        dict[str, str]: {'key': 'value'},
+        dict[str, int]: {'key': 1},
+        dict[str, float]: {'key': 1.0},
+        dict[str, bool]: {'key': True},
```

Comment:
what about union types:

```python
dict[str, Union[str, int]]: ...,
```

Reply:
I added more test cases and this covers even more than I initially thought: https://github.com/openedx/openedx-events/pull/433/commits/a701f78ee5fdc1ae03e2325883c0436b4a65999e. Thanks for the suggestion!